# Strong Concrete with Geopolymer Mortar With/Without Preparation

### Compressive Test

In [16]:
import pandas as pd
from statsmodels.formula.api import ols
from statsmodels.stats.anova import anova_lm

# Import data from CSV file
df = pd.read_csv("D:/Thesis FABGM/SCFABGM/Compressive.csv")

# Fit the two-way ANOVA model
formula = 'Mpa ~ C(Preparation) + C(Mixture) + C(Time) + C(Preparation):C(Mixture) + C(Preparation):C(Time) + C(Mixture):C(Time) + C(Mixture):C(Time):C(Preparation)'
model = ols(formula, df).fit()

# Perform the ANOVA and print the results
anova_table = anova_lm(model)
print(anova_table)


                                     df      sum_sq     mean_sq           F  \
C(Preparation)                      1.0    0.462401    0.462401    0.445134   
C(Mixture)                          2.0   38.863253   19.431626   18.706015   
C(Time)                             3.0  462.572515  154.190838  148.433078   
C(Preparation):C(Mixture)           2.0    0.333553    0.166776    0.160549   
C(Preparation):C(Time)              3.0    6.474015    2.158005    2.077421   
C(Mixture):C(Time)                  6.0   11.575681    1.929280    1.857237   
C(Mixture):C(Time):C(Preparation)   6.0    2.508314    0.418052    0.402441   
Residual                           48.0   49.861933    1.038790         NaN   

                                         PR(>F)  
C(Preparation)                     5.078495e-01  
C(Mixture)                         9.847652e-07  
C(Time)                            2.773939e-24  
C(Preparation):C(Mixture)          8.521318e-01  
C(Preparation):C(Time)             1.1

<div style="text-align: justify">
C(Preparation):
    
    Degrees of Freedom (df): 1.0
    Sum of Squares (sum_sq): 0.462401
    Mean Square (mean_sq): 0.462401
    F-statistic (F): 0.376208
    p-value (PR(>F)): 5.420377e-01 (0.542)
    
    Interpretation: 
    The p-value for "C(Preparation)" is 0.542, which is greater than the significance level (usually 0.05). Thus, there is no significant difference between the "Preparation" groups in terms of the "Mpa" response variable. In other words, the "Preparation" factor does not have a significant effect on the "Mpa."

C(Mixture):
   
    Degrees of Freedom (df): 2.0
    Sum of Squares (sum_sq): 38.863253
    Mean Square (mean_sq): 19.431626
    F-statistic (F): 15.809496
    p-value (PR(>F)): 3.309865e-06 (very close to 0)
    Interpretation: 
    The p-value for "C(Mixture)" is very close to 0, which is less than the significance level. Therefore, there is a significant difference between the "Mixture" groups in terms of the "Mpa" response variable. In other words, the "Mixture" factor has a significant effect on the "Mpa."

C(Time):

    Degrees of Freedom (df): 2.0
    Sum of Squares (sum_sq): 449.152504
    Mean Square (mean_sq): 224.576252
    F-statistic (F): 182.714369
    p-value (PR(>F)): 9.178723e-26 (very close to 0)
    Interpretation:
    The p-value for "C(Time)" is very close to 0, which is less than the significance level. Therefore, there is a significant difference between the "Time" groups in terms of the "Mpa" response variable. In other words, the "Time" factor has a significant effect on the "Mpa."

C(Preparation):C(Mixture), C(Preparation):C(Time), and C(Mixture):C(Time):

    These are the interaction terms between the factors.
    For all three interaction terms, the p-values are greater than the significance level (all above 0.05).
    Interpretation: 
    None of these interaction terms are significant. In other words, there is no significant interaction effect between the combinations of "Preparation" and "Mixture," "Preparation" and "Time," and "Mixture" and "Time" on the "Mpa."
Residual:

    Degrees of Freedom (df): 58.0
    Sum of Squares (sum_sq): 71.288442
    Mean Square (mean_sq): 1.229111
    Interpretation: 
    The residual represents the unexplained variance in the "Mpa" after accounting for the effects of the factors and interactions. It is the variation that cannot be attributed to the factors studied.

    
    
    
    In summary, the results indicate that both "Mixture" and "Time" have significant effects on the "Mpa" response variable. However, the "Preparation" factor and all interactions between factors are not significant.
    
</div>

### Splitting Tensile

In [19]:
import pandas as pd
from statsmodels.formula.api import ols
from statsmodels.stats.anova import anova_lm

# Import data from CSV file
df = pd.read_csv("D:/Thesis FABGM/SCFABGM/Splitting Tensile.csv")

# Fit the two-way ANOVA model
formula = 'Mpa ~ C(Preparation) + C(Mixture) + C(Time) + C(Preparation):C(Mixture) + C(Preparation):C(Time) + C(Mixture):C(Time) + C(Mixture):C(Time):C(Preparation)'
model = ols(formula, df).fit()

# Perform the ANOVA and print the results
anova_table = anova_lm(model)
print(anova_table)


                                     df    sum_sq   mean_sq         F  \
C(Preparation)                      1.0  0.020000  0.020000  0.238584   
C(Mixture)                          2.0  0.086358  0.043179  0.515094   
C(Time)                             3.0  1.974144  0.658048  7.850001   
C(Preparation):C(Mixture)           2.0  0.023908  0.011954  0.142604   
C(Preparation):C(Time)              3.0  0.049167  0.016389  0.195507   
C(Mixture):C(Time)                  6.0  0.050764  0.008461  0.100929   
C(Mixture):C(Time):C(Preparation)   6.0  0.015525  0.002587  0.030867   
Residual                           48.0  4.023733  0.083828       NaN   

                                     PR(>F)  
C(Preparation)                     0.627455  
C(Mixture)                         0.600709  
C(Time)                            0.000231  
C(Preparation):C(Mixture)          0.867463  
C(Preparation):C(Time)             0.898951  
C(Mixture):C(Time)                 0.995942  
C(Mixture):C(Time):C

<div style="text-align: justify">
    
C(Preparation):
    
    Degrees of Freedom (df): 1.0
    Sum of Squares (sum_sq): 0.020000
    Mean Square (mean_sq): 0.020000
    F-statistic (F): 0.267376
    p-value (PR(>F)): 0.607209
    Interpretation: 
    The p-value for "C(Preparation)" is 0.607209, which is greater than the significance level (commonly set at 0.05). Thus, there is no significant difference in Splitting Tensile Strength between different "Preparation" groups. In other words, the "Preparation" factor does not have a statistically significant effect on Splitting Tensile Strength.

C(Mixture):

    Degrees of Freedom (df): 2.0
    Sum of Squares (sum_sq): 0.086358
    Mean Square (mean_sq): 0.043179
    F-statistic (F): 0.577253
    p-value (PR(>F)): 0.564865
    Interpretation: 
    The p-value for "C(Mixture)" is 0.564865, which is greater than the significance level. Therefore, there is no significant difference in Splitting Tensile Strength among the different "Mixture" groups. The "Mixture" factor does not have a statistically significant effect on Splitting Tensile Strength.

C(Time):
    
    Degrees of Freedom (df): 3.0
    Sum of Squares (sum_sq): 1.974144
    Mean Square (mean_sq): 0.658048
    F-statistic (F): 8.797308
    p-value (PR(>F)): 0.000076
    Interpretation: 
    The p-value for "C(Time)" is 0.000076, which is less than the significance level. Therefore, there is a significant difference in Splitting Tensile Strength across the different "Time" groups. The "Time" factor has a statistically significant effect on Splitting Tensile Strength.

C(Preparation):C(Mixture), C(Preparation):C(Time), and C(Mixture):C(Time):

    These are the interaction terms between the factors.
    For all three interaction terms, the p-values are greater than the significance level (all above 0.05).
    Interpretation: 
    None of these interaction terms are significant. In other words, there is no significant interaction effect between the combinations of "Preparation" and "Mixture," "Preparation" and "Time," and "Mixture" and "Time" on Splitting Tensile Strength.

Residual:

    Degrees of Freedom (df): 54.0
    Sum of Squares (sum_sq): 4.039258
    Mean Square (mean_sq): 0.074801
    Interpretation: 
    The residual represents the unexplained variance in Splitting Tensile Strength after accounting for the effects of the factors and interactions. It is the variation that cannot be attributed to the factors studied.

    
    
    
    In summary, the results indicate that the "Time" factor has a significant effect on Splitting Tensile Strength, while the factors "Preparation" and "Mixture" and their interactions do not have significant effects.
    
</div>

### Flexural Strength

In [20]:
import pandas as pd
from statsmodels.formula.api import ols
from statsmodels.stats.anova import anova_lm

# Import data from CSV file
df = pd.read_csv("D:/Thesis FABGM/SCFABGM/Flexural.csv")

# Fit the two-way ANOVA model
formula = 'Mpa ~ C(Preparation) + C(Mixture) + C(Time) + C(Preparation):C(Mixture) + C(Preparation):C(Time) + C(Mixture):C(Time) + C(Mixture):C(Time):C(Preparation)'
model = ols(formula, df).fit()

# Perform the ANOVA and print the results
anova_table = anova_lm(model)
print(anova_table)


                                     df    sum_sq   mean_sq          F  \
C(Preparation)                      1.0  0.931612  0.931612  10.984377   
C(Mixture)                          2.0  1.486108  0.743054   8.761140   
C(Time)                             3.0  2.375938  0.791979   9.338000   
C(Preparation):C(Mixture)           2.0  1.079575  0.539787   6.364480   
C(Preparation):C(Time)              3.0  0.211660  0.070553   0.831873   
C(Mixture):C(Time)                  6.0  0.636825  0.106137   1.251437   
C(Mixture):C(Time):C(Preparation)   6.0  0.303069  0.050512   0.595568   
Residual                           48.0  4.071000  0.084812        NaN   

                                     PR(>F)  
C(Preparation)                     0.001754  
C(Mixture)                         0.000571  
C(Time)                            0.000057  
C(Preparation):C(Mixture)          0.003534  
C(Preparation):C(Time)             0.482943  
C(Mixture):C(Time)                 0.297517  
C(Mixture):

<div style="text-align: justify">
    
C(Preparation):
    
    Degrees of Freedom (df): 1.0
    Sum of Squares (sum_sq): 0.931612
    Mean Square (mean_sq): 0.931612
    F-statistic (F): 11.501206
    p-value (PR(>F)): 0.001307
    Interpretation: 
    The p-value for "C(Preparation)" is 0.001307, which is less than the significance level (commonly set at 0.05). Thus, there is a significant difference in Flexural Strength between different "Preparation" groups. In other words, the "Preparation" factor has a statistically significant effect on Flexural Strength.

C(Mixture):

    Degrees of Freedom (df): 2.0
    Sum of Squares (sum_sq): 1.486108
    Mean Square (mean_sq): 0.743054
    F-statistic (F): 9.173363
    p-value (PR(>F)): 0.000372
    Interpretation: 
    The p-value for "C(Mixture)" is 0.000372, which is less than the significance level. Therefore, there is a significant difference in Flexural Strength among the different "Mixture" groups. The "Mixture" factor has a statistically significant effect on Flexural Strength.

C(Time):
    
    Degrees of Freedom (df): 3.0
    Sum of Squares (sum_sq): 2.375938
    Mean Square (mean_sq): 0.791979
    F-statistic (F): 9.777365
    p-value (PR(>F)): 0.000030
    Interpretation: 
    The p-value for "C(Time)" is 0.000030, which is less than the significance level. Therefore, there is a significant difference in Flexural Strength across the different "Time" groups. The "Time" factor has a statistically significant effect on Flexural Strength.

C(Preparation):C(Mixture), C(Preparation):C(Time), and C(Mixture):C(Time):

    These are the interaction terms between the factors.
    For "C(Preparation):C(Mixture)" and "C(Mixture):C(Time)," the p-values are less than the significance level (both below 0.05).
    However, for "C(Preparation):C(Time)," the p-value is 0.461864, which is greater than 0.05.
    Interpretation: 
    The interactions between "Preparation" and "Mixture" and between "Mixture" and "Time" have statistically significant effects on Flexural Strength. However, the interaction between "Preparation" and "Time" does not have a statistically significant effect on Flexural Strength.

Residual:

    Degrees of Freedom (df): 54.0
    Sum of Squares (sum_sq): 4.374069
    Mean Square (mean_sq): 0.081001
    Interpretation: 
    The residual represents the unexplained variance in Flexural Strength after accounting for the effects of the factors and interactions. It is the variation that cannot be attributed to the factors studied.

    In summary, the results indicate that all main effects of "Preparation," "Mixture," and "Time," as well as the interactions between "Preparation" and "Mixture" and between "Mixture" and "Time," have significant effects on Flexural Strength. However, the interaction between "Preparation" and "Time" does not significantly influence Flexural Strength.
    
</div>

### Bond Strength

In [21]:
import pandas as pd
from statsmodels.formula.api import ols
from statsmodels.stats.anova import anova_lm

# Import data from CSV file
df = pd.read_csv("D:/Thesis FABGM/SCFABGM/Bond.csv")

# Fit the two-way ANOVA model
formula = 'Mpa ~ C(Preparation) + C(Mixture) + C(Time) + C(Preparation):C(Mixture) + C(Preparation):C(Time) + C(Mixture):C(Time) + C(Mixture):C(Time):C(Preparation)'
model = ols(formula, df).fit()

# Perform the ANOVA and print the results
anova_table = anova_lm(model)
print(anova_table)


                                     df     sum_sq   mean_sq         F  \
C(Preparation)                      1.0   3.749235  3.749235  4.990542   
C(Mixture)                          2.0   1.773636  0.886818  1.180428   
C(Time)                             3.0   2.530538  0.843513  1.122785   
C(Preparation):C(Mixture)           2.0   3.067786  1.533893  2.041739   
C(Preparation):C(Time)              3.0   1.218826  0.406275  0.540786   
C(Mixture):C(Time)                  6.0   0.287742  0.047957  0.063835   
C(Mixture):C(Time):C(Preparation)   6.0   0.905436  0.150906  0.200868   
Residual                           48.0  36.060867  0.751268       NaN   

                                     PR(>F)  
C(Preparation)                     0.030180  
C(Mixture)                         0.315904  
C(Time)                            0.349197  
C(Preparation):C(Mixture)          0.140926  
C(Preparation):C(Time)             0.656677  
C(Mixture):C(Time)                 0.998875  
C(Mixture):

<div style="text-align: justify">
    
C(Preparation):
    
    Degrees of Freedom (df): 1.0
    Sum of Squares (sum_sq): 0.931612
    Mean Square (mean_sq): 0.931612
    F-statistic (F): 11.501206
    p-value (PR(>F)): 0.001307
    Interpretation: 
    The p-value for "C(Preparation)" is 0.001307, which is less than the significance level (commonly set at 0.05). Thus, there is a significant difference in Flexural Strength between different "Preparation" groups. In other words, the "Preparation" factor has a statistically significant effect on Flexural Strength.

C(Mixture):

    Degrees of Freedom (df): 2.0
    Sum of Squares (sum_sq): 1.486108
    Mean Square (mean_sq): 0.743054
    F-statistic (F): 9.173363
    p-value (PR(>F)): 0.000372
    Interpretation: 
    The p-value for "C(Mixture)" is 0.000372, which is less than the significance level. Therefore, there is a significant difference in Flexural Strength among the different "Mixture" groups. The "Mixture" factor has a statistically significant effect on Flexural Strength.

C(Time):
    
    Degrees of Freedom (df): 3.0
    Sum of Squares (sum_sq): 2.375938
    Mean Square (mean_sq): 0.791979
    F-statistic (F): 9.777365
    p-value (PR(>F)): 0.000030
    Interpretation: 
    The p-value for "C(Time)" is 0.000030, which is less than the significance level. Therefore, there is a significant difference in Flexural Strength across the different "Time" groups. The "Time" factor has a statistically significant effect on Flexural Strength.

C(Preparation):C(Mixture), C(Preparation):C(Time), and C(Mixture):C(Time):

    These are the interaction terms between the factors.
    For "C(Preparation):C(Mixture)" and "C(Mixture):C(Time)," the p-values are less than the significance level (both below 0.05).
    However, for "C(Preparation):C(Time)," the p-value is 0.461864, which is greater than 0.05.
    Interpretation: 
    The interactions between "Preparation" and "Mixture" and between "Mixture" and "Time" have statistically significant effects on Flexural Strength. However, the interaction between "Preparation" and "Time" does not have a statistically significant effect on Flexural Strength.

Residual:

    Degrees of Freedom (df): 54.0
    Sum of Squares (sum_sq): 4.374069
    Mean Square (mean_sq): 0.081001
    Interpretation: 
    The residual represents the unexplained variance in Flexural Strength after accounting for the effects of the factors and interactions. It is the variation that cannot be attributed to the factors studied.

    In summary, the results indicate that all main effects of "Preparation," "Mixture," and "Time," as well as the interactions between "Preparation" and "Mixture" and between "Mixture" and "Time," have significant effects on Flexural Strength. However, the interaction between "Preparation" and "Time" does not significantly influence Flexural Strength.
    
</div>

### end

# Weak Concrete with Geopolymer Mortar With/Without Preparation

### Compressive Test

In [22]:
import pandas as pd
from statsmodels.formula.api import ols
from statsmodels.stats.anova import anova_lm

# Import data from CSV file
df = pd.read_csv("D:/Thesis FABGM/WCFABGM/Compressive.csv")

# Fit the two-way ANOVA model
formula = 'Mpa ~ C(Preparation) + C(Mixture) + C(Time) + C(Preparation):C(Mixture) + C(Preparation):C(Time) + C(Mixture):C(Time) + C(Mixture):C(Time):C(Preparation)'
model = ols(formula, df).fit()

# Perform the ANOVA and print the results
anova_table = anova_lm(model)
print(anova_table)


                                     df      sum_sq    mean_sq         F  \
C(Preparation)                      1.0    0.783335   0.783335  0.297519   
C(Mixture)                          2.0    3.952144   1.976072  0.750533   
C(Time)                             3.0   55.487082  18.495694  7.024863   
C(Preparation):C(Mixture)           2.0    3.253911   1.626956  0.617935   
C(Preparation):C(Time)              3.0    6.307549   2.102516  0.798558   
C(Mixture):C(Time)                  6.0    1.304656   0.217443  0.082587   
C(Mixture):C(Time):C(Preparation)   6.0    0.808622   0.134770  0.051187   
Residual                           48.0  126.378733   2.632890       NaN   

                                     PR(>F)  
C(Preparation)                     0.587968  
C(Mixture)                         0.477574  
C(Time)                            0.000520  
C(Preparation):C(Mixture)          0.543289  
C(Preparation):C(Time)             0.500775  
C(Mixture):C(Time)                 0.99

<div style="text-align: justify">
    C(Preparation):
    
    Degrees of Freedom (df): 1.0
    Sum of Squares (sum_sq): 0.783335
    Mean Square (mean_sq): 0.783335
    F-statistic (F): 0.297519
    p-value (PR(>F)): 0.587968
    Interpretation: The p-value for "C(Preparation)" is 0.587968, which is greater than the significance level (commonly set at 0.05). Thus, there is no significant difference in Compressive Strength between different "Preparation" groups. In other words, the "Preparation" factor does not have a statistically significant effect on Compressive Strength.

C(Mixture):
    
    Degrees of Freedom (df): 2.0
    Sum of Squares (sum_sq): 3.952144
    Mean Square (mean_sq): 1.976072
    F-statistic (F): 0.750533
    p-value (PR(>F)): 0.477574
    Interpretation: The p-value for "C(Mixture)" is 0.477574, which is greater than the significance level. Therefore, there is no significant difference in Compressive Strength among the different "Mixture" groups. The "Mixture" factor does not have a statistically significant effect on Compressive Strength.

C(Time):
    
    Degrees of Freedom (df): 3.0
    Sum of Squares (sum_sq): 55.487082
    Mean Square (mean_sq): 18.495694
    F-statistic (F): 7.024863
    p-value (PR(>F)): 0.000520
    Interpretation: The p-value for "C(Time)" is 0.000520, which is less than the significance level. Therefore, there is a significant difference in Compressive Strength across the different "Time" groups. The "Time" factor has a statistically significant effect on Compressive Strength.

C(Preparation):C(Mixture), C(Preparation):C(Time), C(Mixture):C(Time), and C(Mixture):C(Time):C(Preparation):

    For all these interaction terms, the p-values are greater than the significance level (all above 0.05).
    Interpretation: None of these interaction terms are significant. In other words, there is no significant interaction effect between the combinations of "Preparation" and "Mixture," "Preparation" and "Time," "Mixture" and "Time," and "Mixture," "Time," and "Preparation" on Compressive Strength.

Residual:

    Degrees of Freedom (df): 48.0
    Sum of Squares (sum_sq): 126.378733
    Mean Square (mean_sq): 2.632890
    Interpretation: The residual represents the unexplained variance in Compressive Strength after accounting for the effects of the factors and interactions. It is the variation that cannot be attributed to the factors studied.

    In summary, the results indicate that the "Time" factor has a significant effect on Compressive Strength. However, the factors "Preparation" and "Mixture" and their interactions with other factors do not have significant effects on Compressive Strength.
    
</div>

### Splitting Tensile

In [23]:
import pandas as pd
from statsmodels.formula.api import ols
from statsmodels.stats.anova import anova_lm

# Import data from CSV file
df = pd.read_csv("D:/Thesis FABGM/WCFABGM/Splitting Tensile.csv")

# Fit the two-way ANOVA model
formula = 'Mpa ~ C(Preparation) + C(Mixture) + C(Time) + C(Preparation):C(Mixture) + C(Preparation):C(Time) + C(Mixture):C(Time) + C(Mixture):C(Time):C(Preparation)'
model = ols(formula, df).fit()

# Perform the ANOVA and print the results
anova_table = anova_lm(model)
print(anova_table)


                                     df    sum_sq   mean_sq          F  \
C(Preparation)                      1.0  0.006235  0.006235   0.604987   
C(Mixture)                          2.0  0.364369  0.182185  17.678302   
C(Time)                             3.0  1.302260  0.434087  42.121608   
C(Preparation):C(Mixture)           2.0  0.087619  0.043810   4.251078   
C(Preparation):C(Time)              3.0  0.005782  0.001927   0.187017   
C(Mixture):C(Time)                  6.0  0.091753  0.015292   1.483872   
C(Mixture):C(Time):C(Preparation)   6.0  0.097681  0.016280   1.579739   
Residual                           48.0  0.494667  0.010306        NaN   

                                         PR(>F)  
C(Preparation)                     4.404988e-01  
C(Mixture)                         1.766971e-06  
C(Time)                            1.727339e-13  
C(Preparation):C(Mixture)          1.996315e-02  
C(Preparation):C(Time)             9.047231e-01  
C(Mixture):C(Time)               

<div style="text-align: justify">
    
    C(Preparation):
    
    Degrees of Freedom (df): 1.0
    Sum of Squares (sum_sq): 0.006235
    Mean Square (mean_sq): 0.006235
    F-statistic (F): 0.604987
    p-value (PR(>F)): 0.4404988
    Interpretation: The p-value for "C(Preparation)" is 0.4404988, which is greater than the significance level (commonly set at 0.05). Thus, there is no significant difference in Splitting Tensile Strength between different "Preparation" groups. In other words, the "Preparation" factor does not have a statistically significant effect on Splitting Tensile Strength.

    C(Mixture):
    
    Degrees of Freedom (df): 2.0
    Sum of Squares (sum_sq): 0.364369
    Mean Square (mean_sq): 0.182185
    F-statistic (F): 17.678302
    p-value (PR(>F)): 1.766971e-06
    Interpretation: The p-value for "C(Mixture)" is 1.766971e-06, which is much less than the significance level. Therefore, there is a significant difference in Splitting Tensile Strength among the different "Mixture" groups. The "Mixture" factor has a statistically significant effect on Splitting Tensile Strength.

    C(Time):
    
    Degrees of Freedom (df): 3.0
    Sum of Squares (sum_sq): 1.302260
    Mean Square (mean_sq): 0.434087
    F-statistic (F): 42.121608
    p-value (PR(>F)): 1.727339e-13
    Interpretation: The p-value for "C(Time)" is 1.727339e-13, which is much less than the significance level. Therefore, there is a significant difference in Splitting Tensile Strength across the different "Time" groups. The "Time" factor has a statistically significant effect on Splitting Tensile Strength.

    C(Preparation):C(Mixture), C(Preparation):C(Time), C(Mixture):C(Time), and C(Mixture):C(Time):C(Preparation):

    For "C(Preparation):C(Mixture)," the p-value is 0.01996315, which is less than the significance level (0.05).
    For all other interaction terms, the p-values are greater than the significance level (all above 0.05).
    Interpretation: The interaction between "Preparation" and "Mixture" has a statistically significant effect on Splitting Tensile Strength. However, the other interaction terms do not significantly influence Splitting Tensile Strength.
    
    Residual:

    Degrees of Freedom (df): 48.0
    Sum of Squares (sum_sq): 0.494667
    Mean Square (mean_sq): 0.010306
    Interpretation: The residual represents the unexplained variance in Splitting Tensile Strength after accounting for the effects of the factors and interactions. It is the variation that cannot be attributed to the factors studied.

        In summary, the results indicate that the "Mixture" and "Time" factors have significant effects on Splitting Tensile Strength. The "Preparation" factor does not have a significant effect. Additionally, there is a significant interaction effect between "Preparation" and "Mixture" but not between other combinations of factors.
    
</div>

### Flexural Strength

In [24]:
import pandas as pd
from statsmodels.formula.api import ols
from statsmodels.stats.anova import anova_lm

# Import data from CSV file
df = pd.read_csv("D:/Thesis FABGM/WCFABGM/Flexural.csv")

# Fit the two-way ANOVA model
formula = 'Mpa ~ C(Preparation) + C(Mixture) + C(Time) + C(Preparation):C(Mixture) + C(Preparation):C(Time) + C(Mixture):C(Time) + C(Mixture):C(Time):C(Preparation)'
model = ols(formula, df).fit()

# Perform the ANOVA and print the results
anova_table = anova_lm(model)
print(anova_table)


                                     df    sum_sq   mean_sq          F  \
C(Preparation)                      1.0  2.753422  2.753422  22.018081   
C(Mixture)                          2.0  4.998503  2.499251  19.985573   
C(Time)                             3.0  2.322417  0.774139   6.190497   
C(Preparation):C(Mixture)           2.0  0.297086  0.148543   1.187843   
C(Preparation):C(Time)              3.0  0.378800  0.126267   1.009707   
C(Mixture):C(Time)                  6.0  0.526775  0.087796   0.702070   
C(Mixture):C(Time):C(Preparation)   6.0  0.536925  0.089487   0.715598   
Residual                           48.0  6.002533  0.125053        NaN   

                                         PR(>F)  
C(Preparation)                     2.278993e-05  
C(Mixture)                         4.848683e-07  
C(Time)                            1.212985e-03  
C(Preparation):C(Mixture)          3.136797e-01  
C(Preparation):C(Time)             3.966537e-01  
C(Mixture):C(Time)               

<div style="text-align: justify">
    
    C(Preparation):
    
    Degrees of Freedom (df): 1.0
    Sum of Squares (sum_sq): 2.753422
    Mean Square (mean_sq): 2.753422
    F-statistic (F): 22.018081
    p-value (PR(>F)): 2.278993e-05
    Interpretation: The p-value for "C(Preparation)" is 2.278993e-05, which is much less than the significance level (commonly set at 0.05). Thus, there is a significant difference in Flexural Strength between different "Preparation" groups. In other words, the "Preparation" factor has a statistically significant effect on Flexural Strength.

    C(Mixture):
    
    Degrees of Freedom (df): 2.0
    Sum of Squares (sum_sq): 4.998503
    Mean Square (mean_sq): 2.499251
    F-statistic (F): 19.985573
    p-value (PR(>F)): 4.848683e-07
    Interpretation: The p-value for "C(Mixture)" is 4.848683e-07, which is much less than the significance level. Therefore, there is a significant difference in Flexural Strength among the different "Mixture" groups. The "Mixture" factor has a statistically significant effect on Flexural Strength.

    C(Time):
    
    Degrees of Freedom (df): 3.0
    Sum of Squares (sum_sq): 2.322417
    Mean Square (mean_sq): 0.774139
    F-statistic (F): 6.190497
    p-value (PR(>F)): 1.212985e-03
    Interpretation: The p-value for "C(Time)" is 1.212985e-03, which is less than the significance level. Therefore, there is a significant difference in Flexural Strength across the different "Time" groups. The "Time" factor has a statistically significant effect on Flexural Strength.

    C(Preparation):C(Mixture), C(Preparation):C(Time), C(Mixture):C(Time), and C(Mixture):C(Time):C(Preparation):

    For all these interaction terms, the p-values are greater than the significance level (all above 0.05).
    Interpretation: None of these interaction terms are significant. In other words, there is no significant interaction effect between the combinations of factors on Flexural Strength.
    
    Residual:

    Degrees of Freedom (df): 48.0
    Sum of Squares (sum_sq): 6.002533
    Mean Square (mean_sq): 0.125053
    Interpretation: The residual represents the unexplained variance in Flexural Strength after accounting for the effects of the factors and interactions. It is the variation that cannot be attributed to the factors studied.

        In summary, the results indicate that the "Preparation" and "Mixture" factors have significant effects on Flexural Strength. Additionally, the "Time" factor also has a significant effect on Flexural Strength. However, there are no significant interaction effects between any combinations of factors on Flexural Strength.
    
</div>

### Bond Strength

In [25]:
import pandas as pd
from statsmodels.formula.api import ols
from statsmodels.stats.anova import anova_lm

# Import data from CSV file
df = pd.read_csv("D:/Thesis FABGM/WCFABGM/Bond.csv")

# Fit the two-way ANOVA model
formula = 'Mpa ~ C(Preparation) + C(Mixture) + C(Time) + C(Preparation):C(Mixture) + C(Preparation):C(Time) + C(Mixture):C(Time) + C(Mixture):C(Time):C(Preparation)'
model = ols(formula, df).fit()

# Perform the ANOVA and print the results
anova_table = anova_lm(model)
print(anova_table)


                                     df     sum_sq   mean_sq         F  \
C(Preparation)                      1.0   0.449668  0.449668  0.417177   
C(Mixture)                          2.0   1.795936  0.897968  0.833086   
C(Time)                             3.0   1.343915  0.447972  0.415604   
C(Preparation):C(Mixture)           2.0   0.049886  0.024943  0.023141   
C(Preparation):C(Time)              3.0   0.057960  0.019320  0.017924   
C(Mixture):C(Time)                  6.0   0.331397  0.055233  0.051242   
C(Mixture):C(Time):C(Preparation)   6.0   0.204469  0.034078  0.031616   
Residual                           48.0  51.738333  1.077882       NaN   

                                     PR(>F)  
C(Preparation)                     0.521425  
C(Mixture)                         0.440893  
C(Time)                            0.742573  
C(Preparation):C(Mixture)          0.977136  
C(Preparation):C(Time)             0.996689  
C(Mixture):C(Time)                 0.999399  
C(Mixture):

<div style="text-align: justify">
    
    C(Preparation):
    
    Degrees of Freedom (df): 1.0
    Sum of Squares (sum_sq): 0.449668
    Mean Square (mean_sq): 0.449668
    F-statistic (F): 0.417177
    p-value (PR(>F)): 0.521425
    Interpretation: The p-value for "C(Preparation)" is 0.521425, which is greater than the significance level (commonly set at 0.05). Thus, there is no significant difference in Bond Strength between different "Preparation" groups. In other words, the "Preparation" factor does not have a statistically significant effect on Bond Strength.

    C(Mixture):
    
    Degrees of Freedom (df): 2.0
    Sum of Squares (sum_sq): 1.795936
    Mean Square (mean_sq): 0.897968
    F-statistic (F): 0.833086
    p-value (PR(>F)): 0.440893
    Interpretation: The p-value for "C(Mixture)" is 0.440893, which is greater than the significance level. Therefore, there is no significant difference in Bond Strength among the different "Mixture" groups. The "Mixture" factor does not have a statistically significant effect on Bond Strength.

    C(Time):
    
    Degrees of Freedom (df): 3.0
    Sum of Squares (sum_sq): 1.343915
    Mean Square (mean_sq): 0.447972
    F-statistic (F): 0.415604
    p-value (PR(>F)): 0.742573
    Interpretation: The p-value for "C(Time)" is 0.742573, which is greater than the significance level. Therefore, there is no significant difference in Bond Strength across the different "Time" groups. The "Time" factor does not have a statistically significant effect on Bond Strength.

    C(Preparation):C(Mixture), C(Preparation):C(Time), C(Mixture):C(Time), and C(Mixture):C(Time):C(Preparation):

    For all these interaction terms, the p-values are greater than the significance level (all above 0.05).
    Interpretation: None of these interaction terms are significant. In other words, there is no significant interaction effect between the combinations of factors on Bond Strength.
    
    Residual:

    Degrees of Freedom (df): 48.0
    Sum of Squares (sum_sq): 51.738333
    Mean Square (mean_sq): 1.077882
    Interpretation: The residual represents the unexplained variance in Bond Strength after accounting for the effects of the factors and interactions. It is the variation that cannot be attributed to the factors studied.

        In summary, the results indicate that neither the "Preparation" nor the "Mixture" factors have a significant effect on Bond Strength. The "Time" factor also does not significantly influence Bond Strength. Additionally, there are no significant interaction effects between any combinations of factors on Bond Strength.
    
</div>

# Strong Concrete with Cement Mortar With/Without Preparation

### Compressive Test

In [26]:
import pandas as pd
from statsmodels.formula.api import ols
from statsmodels.stats.anova import anova_lm

# Import data from CSV file
df = pd.read_csv("D:/Thesis FABGM/SCCM/Compressive.csv")

# Fit the two-way ANOVA model
formula = 'Mpa ~ C(Preparation) + C(Day) + C(Preparation):C(Day)'
model = ols(formula, df).fit()

# Perform the ANOVA and print the results
anova_table = anova_lm(model)
print(anova_table)


                         df      sum_sq    mean_sq          F    PR(>F)
C(Preparation)          1.0    0.598504   0.598504   0.133345  0.719771
C(Day)                  3.0  215.783746  71.927915  16.025276  0.000045
C(Preparation):C(Day)   3.0    0.319046   0.106349   0.023694  0.994859
Residual               16.0   71.814467   4.488404        NaN       NaN


<div style="text-align: justify">
    C(Preparation):
    
    Degrees of Freedom (df): 1.0
    Sum of Squares (sum_sq): 0.598504
    Mean Square (mean_sq): 0.598504
    F-statistic (F): 0.133345
    p-value (PR(>F)): 0.719771
    Interpretation: The p-value for "C(Preparation)" is 0.719771, which is greater than the significance level (commonly set at 0.05). Thus, there is no significant difference in Compressive Strength between different "Preparation" groups. In other words, the "Preparation" factor does not have a statistically significant effect on Compressive Strength.

    C(Day):
    
    Degrees of Freedom (df): 3.0
    Sum of Squares (sum_sq): 215.783746
    Mean Square (mean_sq): 71.927915
    F-statistic (F): 16.025276
    p-value (PR(>F)): 0.000045
    Interpretation: The p-value for "C(Day)" is 0.000045, which is much less than the significance level. Therefore, there is a significant difference in Compressive Strength across the different "Day" groups. The "Day" factor has a statistically significant effect on Compressive Strength.

    C(Preparation):C(Day):
    
    Degrees of Freedom (df): 3.0
    Sum of Squares (sum_sq): 0.319046
    Mean Square (mean_sq): 0.106349
    F-statistic (F): 0.023694
    p-value (PR(>F)): 0.994859
    Interpretation: The p-value for "C(Preparation):C(Day)" is 0.994859, which is greater than the significance level. Therefore, there is no significant interaction effect between the "Preparation" and "Day" factors on Compressive Strength.

    Residual:
    
    Degrees of Freedom (df): 16.0
    Sum of Squares (sum_sq): 71.814467
    Mean Square (mean_sq): 4.488404
    Interpretation: The residual represents the unexplained variance in Compressive Strength after accounting for the effects of the factors and interactions. It is the variation that cannot be attributed to the factors studied.

        In summary, the results indicate that the "Preparation" factor does not have a significant effect on Compressive Strength. However, the "Day" factor has a significant effect on Compressive Strength. There is no significant interaction effect between "Preparation" and "Day" on Compressive Strength.

</div>

### Splitting Tensile

In [27]:
import pandas as pd
from statsmodels.formula.api import ols
from statsmodels.stats.anova import anova_lm

# Import data from CSV file
df = pd.read_csv("D:/Thesis FABGM/SCCM/Splitting Tensile.csv")

# Fit the two-way ANOVA model
formula = 'Mpa ~ C(Preparation) + C(Day) + C(Preparation):C(Day)'
model = ols(formula, df).fit()

# Perform the ANOVA and print the results
anova_table = anova_lm(model)
print(anova_table)


                         df    sum_sq   mean_sq         F    PR(>F)
C(Preparation)          1.0  0.005704  0.005704  0.974377  0.338284
C(Day)                  3.0  0.044579  0.014860  2.538316  0.093194
C(Preparation):C(Day)   3.0  0.001112  0.000371  0.063345  0.978446
Residual               16.0  0.093667  0.005854       NaN       NaN


<div style="text-align: justify">
    
    C(Preparation):
    
    Degrees of Freedom (df): 1.0
    Sum of Squares (sum_sq): 0.005704
    Mean Square (mean_sq): 0.005704
    F-statistic (F): 0.974377
    p-value (PR(>F)): 0.338284
    Interpretation: The p-value for "C(Preparation)" is 0.338284, which is greater than the significance level (commonly set at 0.05). Thus, there is no significant difference in Splitting Tensile Strength between different "Preparation" groups. In other words, the "Preparation" factor does not have a statistically significant effect on Splitting Tensile Strength.

    C(Day):
    
    Degrees of Freedom (df): 3.0
    Sum of Squares (sum_sq): 0.044579
    Mean Square (mean_sq): 0.014860
    F-statistic (F): 2.538316
    p-value (PR(>F)): 0.093194
    Interpretation: The p-value for "C(Day)" is 0.093194, which is slightly greater than the significance level. Therefore, there is no strong evidence to conclude a significant difference in Splitting Tensile Strength across the different "Day" groups. However, the p-value is close to the significance level, so there may be some indication of a potential effect that requires further investigation.

    C(Preparation):C(Day):
    
    Degrees of Freedom (df): 3.0
    Sum of Squares (sum_sq): 0.001112
    Mean Square (mean_sq): 0.000371
    F-statistic (F): 0.063345
    p-value (PR(>F)): 0.978446
    Interpretation: The p-value for "C(Preparation):C(Day)" is 0.978446, which is much greater than the significance level. Therefore, there is no significant interaction effect between the "Preparation" and "Day" factors on Splitting Tensile Strength.

    Residual:
    
    Degrees of Freedom (df): 16.0
    Sum of Squares (sum_sq): 0.093667
    Mean Square (mean_sq): 0.005854
    Interpretation: The residual represents the unexplained variance in Splitting Tensile Strength after accounting for the effects of the factors and interactions. It is the variation that cannot be attributed to the factors studied.

        In summary, the results indicate that the "Preparation" factor does not have a significant effect on Splitting Tensile Strength. The "Day" factor also does not show a strong statistically significant effect, although there may be some potential indication that requires further investigation. There is no significant interaction effect between "Preparation" and "Day" on Splitting Tensile Strength.
    
</div>

### Flexural Strength

In [30]:
import pandas as pd
from statsmodels.formula.api import ols
from statsmodels.stats.anova import anova_lm

# Import data from CSV file
df = pd.read_csv("D:/Thesis FABGM/SCCM/Flexural.csv")

# Fit the two-way ANOVA model
formula = 'Mpa ~ C(Preparation) + C(Day) + C(Preparation):C(Day)'
model = ols(formula, df).fit()

# Perform the ANOVA and print the results
anova_table = anova_lm(model)
print(anova_table)


                         df    sum_sq   mean_sq         F    PR(>F)
C(Preparation)          1.0  0.057038  0.057038  1.298151  0.271322
C(Day)                  3.0  0.268546  0.089515  2.037332  0.149178
C(Preparation):C(Day)   3.0  0.080813  0.026938  0.613087  0.616340
Residual               16.0  0.703000  0.043938       NaN       NaN


<div style="text-align: justify">

    C(Preparation):
    
    Degrees of Freedom (df): 1.0
    Sum of Squares (sum_sq): 0.057038
    Mean Square (mean_sq): 0.057038
    F-statistic (F): 1.298151
    p-value (PR(>F)): 0.271322
    Interpretation: The p-value for "C(Preparation)" is 0.271322, which is greater than the significance level (commonly set at 0.05). Thus, there is no significant difference in Flexural Strength between different "Preparation" groups. In other words, the "Preparation" factor does not have a statistically significant effect on Flexural Strength.

    C(Day):
    
    Degrees of Freedom (df): 3.0
    Sum of Squares (sum_sq): 0.268546
    Mean Square (mean_sq): 0.089515
    F-statistic (F): 2.037332
    p-value (PR(>F)): 0.149178
    Interpretation: The p-value for "C(Day)" is 0.149178, which is greater than the significance level. Therefore, there is no significant difference in Flexural Strength across the different "Day" groups. The "Day" factor does not have a statistically significant effect on Flexural Strength.

    C(Preparation):C(Day):
    
    Degrees of Freedom (df): 3.0
    Sum of Squares (sum_sq): 0.080813
    Mean Square (mean_sq): 0.026938
    F-statistic (F): 0.613087
    p-value (PR(>F)): 0.616340
    Interpretation: The p-value for "C(Preparation):C(Day)" is 0.616340, which is much greater than the significance level. Therefore, there is no significant interaction effect between the "Preparation" and "Day" factors on Flexural Strength.

    Residual:
    
    Degrees of Freedom (df): 16.0
    Sum of Squares (sum_sq): 0.703000
    Mean Square (mean_sq): 0.043938
    Interpretation: The residual represents the unexplained variance in Flexural Strength after accounting for the effects of the factors and interactions. It is the variation that cannot be attributed to the factors studied.

        In summary, the results indicate that neither the "Preparation" nor the "Day" factors have a significant effect on Flexural Strength. Additionally, there is no significant interaction effect between "Preparation" and "Day" on Flexural Strength.
    
</div>

### Bond Strength

In [31]:
import pandas as pd
from statsmodels.formula.api import ols
from statsmodels.stats.anova import anova_lm

# Import data from CSV file
df = pd.read_csv("D:/Thesis FABGM/SCCM/Bond.csv")

# Fit the two-way ANOVA model
formula = 'Mpa ~ C(Preparation) + C(Day) + C(Preparation):C(Day)'
model = ols(formula, df).fit()

# Perform the ANOVA and print the results
anova_table = anova_lm(model)
print(anova_table)


                         df     sum_sq   mean_sq         F    PR(>F)
C(Preparation)          1.0   0.000104  0.000104  0.000246  0.987674
C(Day)                  3.0  11.163412  3.721137  8.796842  0.001121
C(Preparation):C(Day)   3.0   0.072146  0.024049  0.056851  0.981549
Residual               16.0   6.768133  0.423008       NaN       NaN


<div style="text-align: justify">

    C(Preparation):
    
    Degrees of Freedom (df): 1.0
    Sum of Squares (sum_sq): 0.000104
    Mean Square (mean_sq): 0.000104
    F-statistic (F): 0.000246
    p-value (PR(>F)): 0.987674
    Interpretation: The p-value for "C(Preparation)" is 0.987674, which is much greater than the significance level (commonly set at 0.05). Thus, there is no significant difference in Bond Strength between different "Preparation" groups. In other words, the "Preparation" factor does not have a statistically significant effect on Bond Strength.

    C(Day):
    
    Degrees of Freedom (df): 3.0
    Sum of Squares (sum_sq): 11.163412
    Mean Square (mean_sq): 3.721137
    F-statistic (F): 8.796842
    p-value (PR(>F)): 0.001121
    Interpretation: The p-value for "C(Day)" is 0.001121, which is less than the significance level. Therefore, there is a significant difference in Bond Strength across the different "Day" groups. The "Day" factor has a statistically significant effect on Bond Strength.

    C(Preparation):C(Day):
    
    Degrees of Freedom (df): 3.0
    Sum of Squares (sum_sq): 0.072146
    Mean Square (mean_sq): 0.024049
    F-statistic (F): 0.056851
    p-value (PR(>F)): 0.981549
    Interpretation: The p-value for "C(Preparation):C(Day)" is 0.981549, which is much greater than the significance level. Therefore, there is no significant interaction effect between the "Preparation" and "Day" factors on Bond Strength.

    Residual:
    
    Degrees of Freedom (df): 16.0
    Sum of Squares (sum_sq): 6.768133
    Mean Square (mean_sq): 0.423008
    Interpretation: The residual represents the unexplained variance in Bond Strength after accounting for the effects of the factors and interactions. It is the variation that cannot be attributed to the factors studied.

        In summary, the results indicate that the "Preparation" factor does not have a significant effect on Bond Strength. However, the "Day" factor has a significant effect on Bond Strength. There is no significant interaction effect between "Preparation" and "Day" on Bond Strength.
</div>

# Weak Concrete with Cement Mortar With/Without Preparation

### Compressive Strength

In [32]:
import pandas as pd
from statsmodels.formula.api import ols
from statsmodels.stats.anova import anova_lm

# Import data from CSV file
df = pd.read_csv("D:/Thesis FABGM/WCCM/Compressive.csv")

# Fit the two-way ANOVA model
formula = 'Mpa ~ C(Preparation) + C(Day) + C(Preparation):C(Day)'
model = ols(formula, df).fit()

# Perform the ANOVA and print the results
anova_table = anova_lm(model)
print(anova_table)


                         df     sum_sq    mean_sq          F    PR(>F)
C(Preparation)          1.0   0.400417   0.400417   0.265916  0.613136
C(Day)                  3.0  70.039833  23.346611  15.504457  0.000054
C(Preparation):C(Day)   3.0   7.720350   2.573450   1.709025  0.205288
Residual               16.0  24.092800   1.505800        NaN       NaN


<div style="text-align: justify">

    C(Preparation):
    
    Degrees of Freedom (df): 1.0
    Sum of Squares (sum_sq): 0.400417
    Mean Square (mean_sq): 0.400417
    F-statistic (F): 0.265916
    p-value (PR(>F)): 0.613136
    Interpretation: The p-value for "C(Preparation)" is 0.613136, which is much greater than the significance level (commonly set at 0.05). Thus, there is no significant difference in Compressive Strength between different "Preparation" groups. In other words, the "Preparation" factor does not have a statistically significant effect on Compressive Strength.

    C(Day):
    
    Degrees of Freedom (df): 3.0
    Sum of Squares (sum_sq): 70.039833
    Mean Square (mean_sq): 23.346611
    F-statistic (F): 15.504457
    p-value (PR(>F)): 0.000054
    Interpretation: The p-value for "C(Day)" is 0.000054, which is less than the significance level. Therefore, there is a significant difference in Compressive Strength across the different "Day" groups. The "Day" factor has a statistically significant effect on Compressive Strength.

    C(Preparation):C(Day):
    
    Degrees of Freedom (df): 3.0
    Sum of Squares (sum_sq): 7.720350
    Mean Square (mean_sq): 2.573450
    F-statistic (F): 1.709025
    p-value (PR(>F)): 0.205288
    Interpretation: The p-value for "C(Preparation):C(Day)" is 0.205288, which is greater than the significance level. Therefore, there is no significant interaction effect between the "Preparation" and "Day" factors on Compressive Strength.

    Residual:
    
    Degrees of Freedom (df): 16.0
    Sum of Squares (sum_sq): 24.092800
    Mean Square (mean_sq): 1.505800
    Interpretation: The residual represents the unexplained variance in Compressive Strength after accounting for the effects of the factors and interactions. It is the variation that cannot be attributed to the factors studied.

        In summary, the results indicate that the "Preparation" factor does not have a significant effect on Compressive Strength. However, the "Day" factor has a significant effect on Compressive Strength. There is no significant interaction effect between "Preparation" and "Day" on Compressive Strength.
</div>

### Splitting Tensile Strength

In [34]:
import pandas as pd
from statsmodels.formula.api import ols
from statsmodels.stats.anova import anova_lm

# Import data from CSV file
df = pd.read_csv("D:/Thesis FABGM/WCCM/Splitting Tensile.csv")

# Fit the two-way ANOVA model
formula = 'Mpa ~ C(Preparation) + C(Day) + C(Preparation):C(Day)'
model = ols(formula, df).fit()

# Perform the ANOVA and print the results
anova_table = anova_lm(model)
print(anova_table)


                         df    sum_sq   mean_sq         F    PR(>F)
C(Preparation)          1.0  0.003037  0.003037  0.103595  0.751724
C(Day)                  3.0  0.339312  0.113104  3.857468  0.029819
C(Preparation):C(Day)   3.0  0.050212  0.016737  0.570840  0.642270
Residual               16.0  0.469133  0.029321       NaN       NaN


<div style="text-align: justify">

    C(Preparation):
    
    Degrees of Freedom (df): 1.0
    Sum of Squares (sum_sq): 0.003037
    Mean Square (mean_sq): 0.003037
    F-statistic (F): 0.103595
    p-value (PR(>F)): 0.751724
    Interpretation: The p-value for "C(Preparation)" is 0.751724, which is much greater than the significance level (commonly set at 0.05). Thus, there is no significant difference in Splitting Tensile Strength between different "Preparation" groups. In other words, the "Preparation" factor does not have a statistically significant effect on Splitting Tensile Strength.

    C(Day):
    
    Degrees of Freedom (df): 3.0
    Sum of Squares (sum_sq): 0.339312
    Mean Square (mean_sq): 0.113104
    F-statistic (F): 3.857468
    p-value (PR(>F)): 0.029819
    Interpretation: The p-value for "C(Day)" is 0.029819, which is less than the significance level. Therefore, there is a significant difference in Splitting Tensile Strength across the different "Day" groups. The "Day" factor has a statistically significant effect on Splitting Tensile Strength.

    C(Preparation):C(Day):
    
    Degrees of Freedom (df): 3.0
    Sum of Squares (sum_sq): 0.050212
    Mean Square (mean_sq): 0.016737
    F-statistic (F): 0.570840
    p-value (PR(>F)): 0.642270
    Interpretation: The p-value for "C(Preparation):C(Day)" is 0.642270, which is much greater than the significance level. Therefore, there is no significant interaction effect between the "Preparation" and "Day" factors on Splitting Tensile Strength.

    Residual:
    
    Degrees of Freedom (df): 16.0
    Sum of Squares (sum_sq): 0.469133
    Mean Square (mean_sq): 0.029321
    Interpretation: The residual represents the unexplained variance in Splitting Tensile Strength after accounting for the effects of the factors and interactions. It is the variation that cannot be attributed to the factors studied.

        In summary, the results indicate that the "Preparation" factor does not have a significant effect on Splitting Tensile Strength. However, the "Day" factor has a significant effect on Splitting Tensile Strength. There is no significant interaction effect between "Preparation" and "Day" on Splitting Tensile Strength.
</div>

### Flexural Strength

In [35]:
import pandas as pd
from statsmodels.formula.api import ols
from statsmodels.stats.anova import anova_lm

# Import data from CSV file
df = pd.read_csv("D:/Thesis FABGM/WCCM/Flexural.csv")

# Fit the two-way ANOVA model
formula = 'Mpa ~ C(Preparation) + C(Day) + C(Preparation):C(Day)'
model = ols(formula, df).fit()

# Perform the ANOVA and print the results
anova_table = anova_lm(model)
print(anova_table)


                         df    sum_sq   mean_sq         F    PR(>F)
C(Preparation)          1.0  0.180267  0.180267  3.274351  0.089196
C(Day)                  3.0  0.913200  0.304400  5.529100  0.008457
C(Preparation):C(Day)   3.0  0.112600  0.037533  0.681753  0.575998
Residual               16.0  0.880867  0.055054       NaN       NaN


<div style="text-align: justify">
    C(Preparation):
    
    Degrees of Freedom (df): 1.0
    Sum of Squares (sum_sq): 0.180267
    Mean Square (mean_sq): 0.180267
    F-statistic (F): 3.274351
    p-value (PR(>F)): 0.089196
    Interpretation: The p-value for "C(Preparation)" is 0.089196, which is greater than the significance level (commonly set at 0.05). Thus, there is no strong evidence to suggest that there is a significant difference in Flexural Strength between different "Preparation" groups. In other words, the "Preparation" factor does not have a statistically significant effect on Flexural Strength, although it is borderline significant.

    C(Day):
    
    Degrees of Freedom (df): 3.0
    Sum of Squares (sum_sq): 0.913200
    Mean Square (mean_sq): 0.304400
    F-statistic (F): 5.529100
    p-value (PR(>F)): 0.008457
    Interpretation: The p-value for "C(Day)" is 0.008457, which is less than the significance level. Therefore, there is a significant difference in Flexural Strength across the different "Day" groups. The "Day" factor has a statistically significant effect on Flexural Strength.

    C(Preparation):C(Day):
    
    Degrees of Freedom (df): 3.0
    Sum of Squares (sum_sq): 0.112600
    Mean Square (mean_sq): 0.037533
    F-statistic (F): 0.681753
    p-value (PR(>F)): 0.575998
    Interpretation: The p-value for "C(Preparation):C(Day)" is 0.575998, which is much greater than the significance level. Therefore, there is no significant interaction effect between the "Preparation" and "Day" factors on Flexural Strength.

    Residual:
    
    Degrees of Freedom (df): 16.0
    Sum of Squares (sum_sq): 0.880867
    Mean Square (mean_sq): 0.055054
    Interpretation: The residual represents the unexplained variance in Flexural Strength after accounting for the effects of the factors and interactions. It is the variation that cannot be attributed to the factors studied.

        In summary, while the "Preparation" factor does not have a statistically significant effect on Flexural Strength (borderline significant), the "Day" factor has a significant effect. There is no significant interaction effect between "Preparation" and "Day" on Flexural Strength.
    
</div>

### Bond Strength

In [45]:
import pandas as pd
from statsmodels.formula.api import ols
from statsmodels.stats.anova import anova_lm

# Import data from CSV file
df = pd.read_csv("D:/Thesis FABGM/WCCM/Bond.csv")

# Fit the two-way ANOVA model
formula = 'Mpa ~ C(Preparation) + C(Day) + C(Preparation):C(Day)'
model = ols(formula, df).fit()

# Perform the ANOVA and print the results
anova_table = anova_lm(model)
print(anova_table)


                         df    sum_sq   mean_sq         F    PR(>F)
C(Preparation)          1.0  1.161600  1.161600  4.252608  0.055818
C(Day)                  3.0  4.813217  1.604406  5.873716  0.006666
C(Preparation):C(Day)   3.0  0.223567  0.074522  0.272825  0.844092
Residual               16.0  4.370400  0.273150       NaN       NaN


<div style="text-align: justify">
    C(Preparation):
    
    Degrees of Freedom (df): 1.0
    Sum of Squares (sum_sq): 1.161600
    Mean Square (mean_sq): 1.161600
    F-statistic (F): 4.252608
    p-value (PR(>F)): 0.055818
    Interpretation: The p-value for "C(Preparation)" is 0.055818, which is slightly greater than the significance level (commonly set at 0.05). Thus, there is some evidence to suggest that there may be a significant difference in Bond Strength between different "Preparation" groups. However, the p-value is close to the significance level, so it is borderline significant.

    C(Day):
    
    Degrees of Freedom (df): 3.0
    Sum of Squares (sum_sq): 4.813217
    Mean Square (mean_sq): 1.604406
    F-statistic (F): 5.873716
    p-value (PR(>F)): 0.006666
    Interpretation: The p-value for "C(Day)" is 0.006666, which is less than the significance level. Therefore, there is a significant difference in Bond Strength across the different "Day" groups. The "Day" factor has a statistically significant effect on Bond Strength.

    C(Preparation):C(Day):
    
    Degrees of Freedom (df): 3.0
    Sum of Squares (sum_sq): 0.223567
    Mean Square (mean_sq): 0.074522
    F-statistic (F): 0.272825
    p-value (PR(>F)): 0.844092
    Interpretation: The p-value for "C(Preparation):C(Day)" is 0.844092, which is much greater than the significance level. Therefore, there is no significant interaction effect between the "Preparation" and "Day" factors on Bond Strength.

    Residual:
    
    Degrees of Freedom (df): 16.0
    Sum of Squares (sum_sq): 4.370400
    Mean Square (mean_sq): 0.273150
    Interpretation: The residual represents the unexplained variance in Bond Strength after accounting for the effects of the factors and interactions. It is the variation that cannot be attributed to the factors studied.

       In summary, while the "Preparation" factor may have a borderline significant effect on Bond Strength, the "Day" factor has a significant effect. There is no significant interaction effect between "Preparation" and "Day" on Bond Strength.

</div>